src/Backups/RestorerFromBackup.cpp

```diff
@@ -704,9 +705,14 @@ void RestorerFromBackup::checkAccessForObjectsFoundInBackup() const
             insertAtEnd(required_access, access_restorer->getRequiredAccess());
     }
 
-    /// We convert to AccessRights and back to check access rights in a predictable way
-    /// (some elements could be duplicated or not sorted).
-    required_access = AccessRights{required_access}.getElements();
+    /// We convert to AccessRights to check if we have the rights contained in our current access.
+    /// This handles the situation when restoring partial revoked grants:
+    /// GRANT SELECT ON *.*
+    /// REVOKE SELECT FROM systems.*
+    const auto & required_access_rights = AccessRights{required_access};
+    const auto & access_rights = context->getAccess()->getAccessRights();
+    if (access_rights->contains(required_access_rights))
+        return;
 
     context->checkAccess(required_access);
```

Comment:
Maybe we could make `contains` a part of `checkAccess`, WDYT?

Reply:
I tried implementing this, but the code change has grown beyond the current PR. I will continue the refactoring in the separate PR, but this required a complete rewriting of `checkAccessImplHelper` and some other functions from checking access per flag to check access per access_rights set